packages/react-native-hw-transport-ble/src/BleTransport.ts

```diff
@@ -445,6 +461,9 @@ export default class BluetoothTransport extends Transport {
         mtu =
           (await merge(
             this.notifyObservable.pipe(
+              tap((maybeError) => {
+                if (maybeError instanceof Error) throw maybeError;
```

Comment:
are you sure throwing from a `tap` works? It feels a bit weird but good to know if it does, it makes the underlying observable failing?

that said, this part really feels weird to me, like why would an event of the notify be an instance of Error, we are making the type "any" here which isn't great either. maybe it's a `Buffer|PairingFailed` to be more specific?

Reply:
I'm sure it works (the tap throw) because I can handle the error that was crashing the app.
It all **seems weird to me too!** yes, but I was 100% unable to catch the error otherwise, the error is thrown inside the monitor method, so inside the ble-plx, and if we don't handle it this way. Ie, if we throw from outside the transport itself, then it's unhandled on the UI and crashes the app.

As for the typing I couldn't get it to like exactly that `Buffer|PairingFailed` but if you think it can work, I'm open to suggestions.